### PR TITLE
Added parsing of variable.intermediateUpdate

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

File: Test/FMI3/fmi3_import_variable_test.cpp [hunk 1]
```diff
@@ -268,8 +271,18 @@ static void test_clock_all_attrs(fmi3_import_t* xml) {
     REQUIRE(fmi3_import_get_clock_variable_shift_counter(cv)        == 6);
 }
 
+/**
+ * Tests parsing a variable with non-default canHandleMultipleSetPerTimeInstant attribute.
+ */
+static void test_non_default_canhandlemultiplesetpertimeinstant(fmi3_import_t* xml) {
```
Comment (PeterMeisrimelModelon) [inline Test/FMI3/fmi3_import_variable_test.cpp:277]: Did not have a test for this before.

File: Test/FMI3/parser_test_xmls/variable_test/valid/basic/modelDescription.xml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <fmiModelDescription fmiVersion="3.0" modelName="" instantiationToken="">
-<ModelExchange modelIdentifier="id"/>
+<ModelExchange modelIdentifier="id_me"/>
+<CoSimulation modelIdentifier="id_cs"/>
```
Comment (PeterMeisrimelModelon) [inline Test/FMI3/parser_test_xmls/variable_test/valid/basic/modelDescription.xml:4]: Without Co-Simulation, intermediateUpdate is ignored.

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 3]
```diff
@@ -1254,6 +1270,47 @@ static int fmi3_xml_variable_process_attr_multipleset(fmi3_xml_parser_context_t*
...
+
+    // peek due to "must not have attribute" error check
+    if (!fmi3_xml_peek_attr_str(context, fmi_attr_id_intermediateUpdate)) { // if attribute not set, use default
+        variable->intermediateUpdate = 0; // default
+        return 0;
+    } 
+    // else: attribute exists
+
+    fmi3_fmu_kind_enu_t fmuKind = fmi3_xml_get_fmu_kind(md);
+    // Spec: "This attribute is ignored in Model Exchange and Scheduled Execution", ignored unless Co-Simulation
```
Comment (PeterMeisrimelModelon) [inline src/XML/src/FMI3/fmi3_xml_variable.c:1287]: If ignored, should we still set the default value?
This attribute is only relevant for Co-Simulation.

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 4]
```diff
@@ -1254,6 +1270,47 @@ static int fmi3_xml_variable_process_attr_multipleset(fmi3_xml_parser_context_t*
...
+    // Multiple types can be defined, check for: not CS
+    if (!(fmuKind & fmi3_fmu_kind_cs)) {
+        jm_log_info(md->callbacks, "FMI3XML", "'intermediateUpdate' attribute ignored since FMU kind is not Co-Simulation.");
+        return 0;
+    }
+
+    // Spec: "Variables of type Clock must not have the intermediateUpdate attribute"
```
Comment (PeterMeisrimelModelon) [inline src/XML/src/FMI3/fmi3_xml_variable.c:1294]: I interpret "must not have" such that even Clock variable with intermediateUpdate = <default> would be invalid. Hence the peek

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 5]
```diff
@@ -1254,6 +1270,47 @@ static int fmi3_xml_variable_process_attr_multipleset(fmi3_xml_parser_context_t*
...
+        return -1;
+    }
+
+    if (fmi3_xml_set_attr_boolean(context, elm_id, fmi_attr_id_intermediateUpdate,
+            0 /* required */, &intermediateUpdate, 0 /* defaultVal */)) {
+        return -1;
+    }
+    variable->intermediateUpdate = (char)intermediateUpdate;
+
+    // Spec: "Variables with causality = parameter must not be marked with intermediateUpdate = true"
```
Comment (PeterMeisrimelModelon) [inline src/XML/src/FMI3/fmi3_xml_variable.c:1306]: I think it should also include structural & calculatedParameters and I am considering asking for clarification in the standard. 

Any input appreciated.

See for reference
https://fmi-standard.org/docs/3.0/#figure-co-simulation-state-machine

`intermediateUpdate`: variable accessible during Intermediate Update mode.
However, the modes when structuralParameters & calculatedParameters do not include Intermediate Update mode, see https://fmi-standard.org/docs/3.0/#table-variableBase-attributes

**Review by filip-stenstrom — CHANGES_REQUESTED** [review]
Good, minor comments

File: src/Import/include/FMI3/fmi3_import_variable.h [hunk 6]
```diff
@@ -203,6 +203,11 @@ FMILIB_EXPORT fmi3_import_variable_t* fmi3_import_get_previous(fmi3_import_varia
 */
 FMILIB_EXPORT fmi3_boolean_t fmi3_import_get_canHandleMultipleSetPerTimeInstant(fmi3_import_variable_t* v);
 
+/** \brief Get the intermediateUpdate flag for a variable.
+
+    @return If true, the variable can be accessed in Intermediate Update Mode
+*/
+FMILIB_EXPORT fmi3_boolean_t fmi3_import_get_intermediateUpdate(fmi3_import_variable_t* v);
```
[the remark this answers is not in the document]
Comment (filip-stenstrom) [reply src/Import/include/FMI3/fmi3_import_variable.h]: Okay sure. Was considering it as part of general symbol renaming to snake_case. We had a few other ones as well.

File: src/XML/src/FMI3/fmi3_xml_variable.c (hunk 3 above)
Comment (filip-stenstrom) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1282]: I don't think you should check for ignore before peeking. Won't you now give this Info message on all variables even when the attribute is not defined?

File: src/XML/src/FMI3/fmi3_xml_variable.c (hunk 4 above)
Comment (filip-stenstrom) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1296]: Agree

File: src/XML/src/FMI3/fmi3_xml_variable.c (hunk 3 above)
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1287]: Good catch, fixed and adjusted the order.

**Review by filip-stenstrom — APPROVED** [review]

**Pull request merged**